resources/wikia/modules/importScript.js

```diff
@@ -0,0 +1,52 @@
+define('wikia.importScript', ['wikia.importScriptHelper', 'wikia.window'], function(importScriptHelper, window) {
+	'use strict';
+
+	/**
+	 * Imports script from provided JS page name
+	 * Page has to be in MediaWiki namespace and has .js extension
+	 *
+	 * Article name can point a page on other wikia by db name
+	 * (e.g. 'external:otherwikiadatabasename:Pagename.js')
+	 *
+	 * or by subdomain or host
+	 * (e.g. 'url:otherwikiasubdomain:Pagename.js')
+	 * (e.g. 'url:otherwikiasubdomain.wikia.com:Pagename.js')
+	 *
+	 * @param {array} articles Names of pages to import without namespace prefix
+	 */
+	function importWikiaScriptPages(articles) {
+		var articlesToImport = [],
+			articlesFailed = [],
+			scriptName;
+
+		if (!$.isArray(articles)) {
+			articles = [articles];
+		}
+
+		for (var i = 0; i < articles.length; i++) {
+			if (!importScriptHelper.isJsPage(articles[i])) {
+				articlesFailed.push(namespacePrefix + articles[i]);
+				continue;
+			}
+			if (importScriptHelper.isLocal(articles[i])) {
+				articlesToImport.push(importScriptHelper.getNamespacePrefix() + articles[i]);
```

Comment:
In this case, I'd consider changing the user-facing message that currently is: ```{script name} was not loaded as it's not a JS page (requested by user-supplied javascript). Please make sure the article title has .js extension.``` which is misleading.